test/integration/029_docs_generate_tests/test_docs_generate.py

```diff
@@ -3180,3 +3184,65 @@ def test_postgres_override_used(self):
             self.run_dbt(['docs', 'generate'])
 
         self.assertIn('rejected: no catalogs for you', str(exc.exception))
+
+
+@mark.skipif(os.name != 'nt', reason='This is only relevant on windows')
+class TestDocsGenerateLongWindowsPaths(DBTIntegrationTest):
+    def _generate_test_root_dir(self):
+        assert os.name == 'nt'
+        magic_prefix = '\\\\?\\'
+
+        # tempfile.mkdtemp doesn't use `\\?\` by default so we have to
+        # get a tiny bit creative.
+        temp_dir = tempfile.gettempdir()
+        if not temp_dir.startswith(magic_prefix):
+            temp_dir = magic_prefix + temp_dir
+        outer = tempfile.mkdtemp(prefix='dbt-int-test-', dir=temp_dir)
+        # then inside _that_ directory make a new one that gets us to just
+        # barely 260 total. I picked 250 to account for the '\' and anything
```

Comment:
is there a reason we want to just barely eclipse 260 chars instead of going for, say, 300 chars? I just want to make sure I understand the logic behind this test.

Reply:
Yeah! We have to `os.chdir` into the directory we build here, and the `\\?\` doesn't work for `os.chdir` because the underlying windows APIs for that function don't handle it.

Reply:
thanks, i hate it